src/hooks/mongodb.js

```diff
@@ -1,98 +1,61 @@
-import { SpansContainer, TracerGlobals } from '../globals';
+import { hook } from '../extender';
 import * as logger from '../logger';
-import { getCurrentTransactionId } from '../spans/awsSpan';
-import { createMongoDbSpan, extendMongoDbSpan } from '../spans/mongoDbSpan';
-import { getRandomId, safeExecute } from '../utils';
+import { safeExecute } from '../utils';
 import { safeRequire } from '../utils/requireUtils';
+import { onFailedHook, onStartedHook, onSucceededHook } from './mongodb3x';
+import { afterConstructorHook, beforeConstructorHook } from './mongodb4x';
 
-const PendingRequests = (() => {
-  let pendingRequests = {};
-
-  const addPendingRequest = (requestId, spanId) => {
-    pendingRequests[requestId] = spanId;
-  };
-
-  const popPendingRequestSpanId = (requestId) => {
-    const spanId = pendingRequests[requestId];
-    delete pendingRequests[requestId];
-    return spanId;
-  };
+export const hookMongoDb = (mongoClientLibrary) => {
+  const mongoClientLibraries = mongoClientLibrary ? mongoClientLibrary : safeRequire('mongodb');
+  const mongooseClients = safeRequire('node_modules/mongoose/node_modules/mongodb');
+  const mongoClients = [mongoClientLibraries, mongooseClients].filter(Boolean);
 
-  return {
-    addPendingRequest,
-    popPendingRequestSpanId,
-  };
-})();
+  /* istanbul ignore next : there's no point in testing this */
+  if (!mongoClients) {
+    logger.debug('MongoDB clients not found');
+    return;
+  }
 
-const onStartedHook = (event) => {
-  const awsRequestId = TracerGlobals.getHandlerInputs().context.awsRequestId;
-  const transactionId = getCurrentTransactionId();
-  const { command, databaseName, commandName, requestId, operationId, connectionId } = event;
-  const started = Date.now();
-  const spanId = getRandomId();
-  PendingRequests.addPendingRequest(requestId, spanId);
-  const mongoSpan = createMongoDbSpan(
-    transactionId,
-    awsRequestId,
-    spanId,
-    {
-      started,
-    },
-    {
-      databaseName,
-      commandName,
-      command,
-      mongoRequestId: requestId,
-      mongoOperationId: operationId,
-      mongoConnectionId: connectionId,
+  mongoClients.forEach((mongoClientLibrary) => {
+    let mongoClientVersion = 'unknown';
+    if (mongoClientLibrary.instrument) {
+      mongoClientVersion = '3.x';
+    } else if (mongoClientLibrary.MongoClient.prototype.on) {
+      mongoClientVersion = '4.x';
+    }
```

Comment:
Try to clean up the code; instead of doing this if only for a switch case, you can do then in once
```
if (mongoClientLibrary.instrument) {
    instrumentMongo3x()
}
if (mongoClientLibrary.MongoClient.prototype.on) {
     instrumentMongo4x()
}
```

Reply:
i feel that moving it out is more readable, i think making the version detection a separate function will be cleaner and more readable so i'll try that instead